clients/java/src/main/java/io/unitycatalog/client/ApiClientBuilder.java

```diff
@@ -0,0 +1,143 @@
+package io.unitycatalog.client;
+
+import io.unitycatalog.client.auth.TokenProvider;
+import io.unitycatalog.client.internal.Preconditions;
+import io.unitycatalog.client.internal.RetryingApiClient;
+import io.unitycatalog.client.retry.JitterDelayRetryPolicy;
+import io.unitycatalog.client.retry.RetryPolicy;
+import java.net.URI;
+
+/**
+ * Builder to create configured {@link ApiClient} instances for Unity Catalog operations.
+ *
+ * <p>Example usage:
+ *
+ * <pre>{@code
+ * ApiClient client = ApiClientBuilder.create()
+ *     .url("http://localhost:8080")
+ *     .tokenProvider(TokenProvider.create("my-token"))
+ *     .retryPolicy(JitterDelayRetryPolicy.builder().maxAttempts(5).build())
+ *     .clientVersion("MyApp", "1.0.0", "Java", "11")
+ *     .build();
+ * }</pre>
+ *
+ * @see ApiClient
+ * @see TokenProvider
+ * @see RetryPolicy
+ */
+public class ApiClientBuilder {
+  private static final String BASE_PATH = "/api/2.1/unity-catalog";
+
+  private URI uri = null;
+  private TokenProvider tokenProvider = null;
+  private String[] nameVersionPairs = null;
+  private RetryPolicy retryPolicy = JitterDelayRetryPolicy.builder().build();
+
+  /**
+   * Creates a new instance of {@link ApiClientBuilder}.
+   *
+   * @return a new {@link ApiClientBuilder} instance
+   */
+  public static ApiClientBuilder create() {
+    return new ApiClientBuilder();
+  }
+
+  /**
+   * Sets the Unity Catalog server URI.
+   *
+   * <p>The base path {@code /api/2.1/unity-catalog} will be automatically appended.
+   *
+   * @param uri the Unity Catalog server URI, must not be null
+   * @return this builder instance for method chaining
+   */
+  public ApiClientBuilder uri(URI uri) {
+    this.uri = uri;
+    return this;
+  }
+
+  /**
+   * Sets the Unity Catalog server URI from a string.
+   *
+   * @param uri the Unity Catalog server URI, must not be null
+   * @return this builder instance for method chaining
+   * @see #uri(URI)
+   */
+  public ApiClientBuilder uri(String uri) {
+    return uri(URI.create(uri));
+  }
+
+  /**
+   * Sets the token provider for authentication.
+   *
+   * <p>The token will be included in the Authorization header as a Bearer token.
+   *
+   * @param tokenProvider the token provider implementation, must not be null
+   * @return this builder instance for method chaining
+   * @see TokenProvider
+   */
+  public ApiClientBuilder tokenProvider(TokenProvider tokenProvider) {
+    this.tokenProvider = tokenProvider;
+    return this;
+  }
+
+  /**
+   * Sets application version metadata as name-version pairs.
```

Comment:
add to the docs that this appversion is set as the user agent using this format - https://www.rfc-editor.org/rfc/rfc9110#name-user-agent